src/base/Assert.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_BASE_ASSERT_H
+#define SQUID_SRC_BASE_ASSERT_H
+
+#include "base/Here.h"
+
+/// Reports the described assertion (at the given debugging level) and throws
+/// the corresponding exception. Reduces compiled code size of Assert() and
+/// Must() callers. Do not call directly; use Assert() instead.
+/// \param description explains the condition (i.e. what MUST happen)
+[[ noreturn ]] void ReportAndThrow_(int debugLevel, const char *description, const SourceLocation &);
```

Comment:
"report" how to where, etc ?
```suggestion
[[ noreturn ]] void LogAndThrow_(int debugLevel, const char *description, const SourceLocation &);
```